Update/fix migration docs for SSP to ES. (#81)
canonical-ols/snapstore-snap-docs: docs/how-to/migrate.md

directory to the Enterprise Store’s common directory:

```
sudo cp -r /var/snap/snap-store-proxy/common/snapstorage-local /var/snap/enterprise-store/common/snapstorage-local
sudo cp -r /var/snap/snap-store-proxy/common/nginx/airgap /var/snap/enterprise-store/common/nginx/airgap
sudo cp -r /var/snap/snap-store-proxy/common/snaps-to-push /var/snap/enterprise-store/common/snaps-to-push
sudo cp -r /var/snap/snap-store-proxy/common/charms-to-push /var/snap/enterprise-store/common/charms-to-push
```

3. Export the existing configuration with:

```
sudo snap-store-proxy config --export-yaml | cat > store-config.yaml
```

```{warning}
The YAML file should be stored securely, since it includes sensitive data like secrets.
```

```{note}
If the `--export-yaml` option is unavailable, upgrade to the latest
version of the `snap-store-proxy` snap, which will contain the
`--export-yaml` option.
```

4. Import the configuration to the Enterprise Store:

```
cat store-config.yaml | sudo enterprise-store config --import-yaml
```

This command may fail, since the Enterprise Store service ports
conflict with the existing Snap Store Proxy service ports. This should

> be expected. An example output of this is:

Reviewer: the config is imported with the expected failure, right? maybe we can add sth like "This should
be expected and the configuration is still imported"

Author: Yep accepted a suggestion from SecondSkoll that should address that

> This command may fail, since the Enterprise Store service ports
> conflict with the existing Snap Store Proxy service ports. This should
> be expected. An example output of this is:

Reviewer 2: ```suggestion
**This command will fail** if Snap Store Proxy is running, as the Enterprise
Store service ports conflict with the existing Snap Store Proxy service ports.
This is expected. An example output of this is:
```

Reviewer 2: Certainty of failure under specific context, rather than possible failure / success. Bold to highlight this - so it's less likely to be skipped / ignored (could also make it into an admonition but it's pretty large to do that).

> ```bash
> $ cat store-config.yaml | sudo enterprise-store config --import-yaml
> Configured database for packagereview role.
> Configured database for packagereview-celery role.
> Configured database for snaprevs role.
> Configured database for snapauth role.
> Configured database for snapident role.
> Configured database for snapassert role.
> Configured database for updown role.
> Configured database for snapmodels role.
> error: Command '['snapctl', 'restart', '--reload', 'enterprise-store.nginx', 'enterprise-store.snapmodels', 'enterprise-store.memcached', 'enterprise-store.storeadmingw', 'enterprise-store.packagereview', 'enterprise-store.snapassert', 'enterprise-store.snapauth', 'enterprise-store.snapproxy', 'enterprise-store.snapstorage', 'enterprise-store.packagereview-worker', 'enterprise-store.snapident', 'enterprise-store.snaprevs', 'enterprise-store.snapdevicegw', 'enterprise-store.publishergw']' returned non-zero exit status 1.
> ```

Reviewer 2: ````suggestion
```{terminal}
:input: cat store-config.yaml | sudo enterprise-store config --import-yaml
:copy:

Configured database for packagereview role.
Configured database for packagereview-celery role.
Configured database for snaprevs role.
Configured database for snapauth role.
Configured database for snapident role.
Configured database for snapassert role.
Configured database for updown role.
Configured database for snapmodels role.
error: Command '['snapctl', 'restart', '--reload', 'enterprise-store.nginx', 'enterprise-store.snapmodels', 'enterprise-store.memcached', 'enterprise-store.storeadmingw', 'enterprise-store.packagereview', 'enterprise-store.snapassert', 'enterprise-store.snapauth', 'enterprise-store.snapproxy', 'enterprise-store.snapstorage', 'enterprise-store.packagereview-worker', 'enterprise-store.snapident', 'enterprise-store.snaprevs', 'enterprise-store.snapdevicegw', 'enterprise-store.publishergw']' returned non-zero exit status 1.
```
````

Reviewer 2: Part of our guidance around code-blocks is to separate input from output. We have an extension that handles that if we use it - though it's not really necessary. Just a QOL thing.

> 5. Use the Enterprise Store's services instead of the Snap Store Proxy's.

Reviewer 2: ```suggestion
5. Replace the Snap Store Proxy's services with the configured Enterprise Store.
```

Disable the Snap Store Proxy snap (this may cause temporary downtime):

```
sudo snap disable snap-store-proxy
```

Start the Enterprise Store services:

```
sudo snap start enterprise-store
```

> The Enterprise Store services should now be able to bind to the
> appropriate ports. Check the status:

Reviewer 2: ```suggestion
Check the status:
```

We provided context on the port conflict earlier - no need to revisit in a how-to (or if you feel strongly, add a note as a separate admonition after the sample status output).
